packages/components/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,225 @@
+import React, {
+  useEffect,
+  useState,
+  useRef,
+  MouseEvent,
+  FocusEvent,
+  CSSProperties,
+} from 'react';
+import { usePopper } from 'react-popper';
+import { Placement } from '@popperjs/core';
+import cn from 'classnames';
+import type * as CSS from 'csstype';
+
+import { styles } from './Tooltip.styles';
+
+import tokens from '@contentful/f36-tokens';
+
+import { Portal } from '@contentful/f36-utils';
+
+export type TooltipPlace = Placement;
+
+export interface TooltipProps {
+  /**
+   * Child nodes to be rendered in the component and that will show the tooltip when they are hovered
+   */
+  children: React.ReactNode;
+  /**
+   * Class names to be appended to the className prop of the Tooltip wrapper
+   */
+  className?: string;
+  /**
+   * HTML element used to wrap the target of the Tooltip
+   */
+  as?: React.ElementType;
+  /**
+   * Content of the Tooltip
+   */
+  content?: React.ReactNode;
+  id?: string;
+  /**
+   * It controls the initial visibility of the Tooltip
+   */
+  isVisible?: boolean;
+  /**
+   * It sets a max-width for the Tooltip
+   */
+  maxWidth?: number | CSS.Property.MaxWidth;
+  /**
+   * It sets a delay period for the Tooltip
+   */
+  hideDelay?: number;
+  /**
+   * Function that will be called when target gets blurred
+   */
+  onBlur?: (evt: FocusEvent) => void;
+  /**
+   * Function that will be called when target gets focused
+   */
+  onFocus?: (evt: FocusEvent) => void;
+  /**
+   * Function that will be called when the user move the mouse out of the target
+   */
+  onMouseLeave?: (evt: MouseEvent) => void;
+  /**
+   * Function that will be called when the user move the mouse over of the target
+   */
+  onMouseOver?: (evt: MouseEvent) => void;
+  /**
+   * It sets the "preferred" position of the Tooltip
+   */
+  place?: TooltipPlace;
+  /**
+   * Class names to be appended to the className prop of the Tooltip’s target
+   */
+  targetWrapperClassName?: string;
+  /**
+   * An ID used for testing purposes applied as a data attribute (data-test-id)
+   */
+  testId?: string;
+  /**
+   * Boolean to control whether or not to render the tooltip in a React Portal.
+   * Rendering content inside a Portal allows the tooltip to escape the bounds
+   * of its parent while still being positioned correctly. Using a Portal is
+   * necessary if an ancestor of the tooltip hides overflow.
+   *
+   * Defaults to `false`
+   */
+  usePortal?: boolean;
+}
+
+export const Tooltip = ({
+  children,
+  className,
+  as: HtmlTag = 'span',
+  content,
+  id,
+  isVisible = false,
+  hideDelay = 0,
+  onBlur,
+  onFocus,
+  onMouseLeave,
+  onMouseOver,
+  targetWrapperClassName,
+  maxWidth = 360,
+  testId = 'cf-ui-tooltip',
+  place = 'auto',
```

Comment:
I'd probably rename it to `placement`. What do you think?

Reply:
yeah, placement sounds much better